### ISSUE TITLE
'The handle is invalid' error when importing audio with the .exe file

### ISSUE THREAD

**Issue by Desati123** [Corne2Plum3/fnf2osumania#3]
**Describe the bug**
Not too sure what happened here but after plugging in all of the files, it just didn't want to export the .osz

     **Error message**

Traceback (most recent call last):
  File "fnf_converter.py", line 505, in exportAsOsz
  File "site-packages\pydub\audio_segment.py", line 685, in from_file
  File "site-packages\pydub\utils.py", line 274, in mediainfo_json
  File "subprocess.py", line 753, in __init__
  File "subprocess.py", line 1054, in _get_handles
OSError: [WinError 6] The handle is invalid
   
    **Screenshots**

![thingy](https://user-images.githubusercontent.com/53101674/135735485-048fa455-a5f6-423f-b836-c8aab215b8d8.PNG)
   
    **OS and version:**

Windows 10
v1.1 (Latest)

**Comment by Desati123** [comment]
I thought about it for a second, and I just realized the song I'm trying to upload has many special custom notes (For instance ones that drain your health if you don't press them) obviously osu doesn't have notes like that in mania so is there some sort of incompatibility going on here? 🤔

**Comment by Desati123** [comment]
I tried once again to no avail, this time with the AGOTI mod. I know for a fact that it shouldn't have any custom notes yet I still got this error message:
Traceback (most recent call last):
  File "fnf_converter.py", line 505, in exportAsOsz
  File "site-packages\pydub\audio_segment.py", line 685, in from_file
  File "site-packages\pydub\utils.py", line 274, in mediainfo_json
  File "subprocess.py", line 753, in __init__
  File "subprocess.py", line 1054, in _get_handles
OSError: [WinError 6] The handle is invalid

**Comment by Marcos875433** [comment]
I've found out something curious, if I use the v1.1 from the release it gets the same error as yours, exactly the same, but if I build it from the source code, it's slightly different, see: 
```
Traceback (most recent call last):
File "fnf_converter.py", line 505, in exportAsOsz
File "pydub\audio_segment.py", line 728, in from_file
  File "pydub\utils.py", line 274, in mediainfo_json
  File "subprocess.py", line 808, in __init__
  File "subprocess.py", line 1146, in _get_handles
OSError: [WinError 6] The handle is invalid
```
you may notice that the lines where the error happens is different from yours, so this is what I did:
I followed the instructions to build the source code and it didn't work at first, I got an error that said something like "Module not found: pydub" at the end of the whole thing. So I installed pydub and retried, and it did work, but when I tried to convert from fnf to osu it threw me the error from above, then I tried installing all the dependencies for developing, and rebuid, it didn't work too, and nothing changed.

**Comment by Corne2Plum3** [comment]
> I thought about it for a second, and I just realized the song I'm trying to upload has many special custom notes (For instance ones that drain your health if you don't press them) obviously osu doesn't have notes like that in mania so is there some sort of incompatibility going on here? 🤔

This is not related to the chart since at this moment the program didn't checked the chart file...

It's a problem which happens when you import the audio... I don't get it how it happens now, but several people reported the issue.
I will check that...

**Comment by tiaomafy** [comment]
that problem is caused from a error of creating audio.mp3.
i managed to fix that from myself, but you have to do the audio from yourself in audacity, mixing the inst and voices
to skip the part of the audio.mp3, edit the fnf_converter.py and delete the script from line 503 to 517
thats what i did to avoid that issue
you have to put the audio.mp3 in the folder in .osz, to open the .osz has a zip file just replace the .osz to .zip in the name

**Comment by Corne2Plum3** [comment]
> that problem is caused from a error of creating audio.mp3. i managed to fix that from myself, but you have to do the audio from yourself in audacity, mixing the inst and voices to skip the part of the audio.mp3, edit the fnf_converter.py and delete the script from line 503 to 517 thats what i did to avoid that issue you have to put the audio.mp3 in the folder in .osz, to open the .osz has a zip file just replace the .osz to .zip in the name

Your idea is basically skipping lines that making the bug and do them manually... Yeah it works sure but it's weird to do that...

After some tests (I'm busy so sorry if I take time), I have the bug only with the .exe file and not with the source code...

But some people already encountered this kind of issue when they convert the program to .exe... I still didn't find a fix now...

**Comment by Desati123** [comment]
How do you use the source code build on windows?

I'm not sure how to open .py files and when I open the main.py or fnf_converter.py nothing happened

**Comment by Desati123** [comment]
I tried to open the source code after installing all the dependencies with pip, but I only get these error messages
![1](https://user-images.githubusercontent.com/53101674/135779527-a3ed8094-b139-498f-8849-6fe8eccbedca.PNG)
![2](https://user-images.githubusercontent.com/53101674/135779528-9c4ac458-7fb2-4e8e-b829-1fcfb5a749dd.PNG)
:

**Comment by Marcos875433** [comment]
ok, I was experimenting and it worked from the source code, so what you have to do is go to the root folder where the source code is located, and in the path direction, type `cmd` and hit enter, after that a window will pop up, and in that window you have to type `python3 main.py` and for some reason it works and converting fnf to osu also works, so it works but only from the source code

**Comment by Desati123** [comment]
After learning you could open the command prompt by typing in cmd to the directory search bar, the program works! I shouldn't close this thread but thanks so very much for the advice!

**Comment by Marcos875433** [comment]
I found out a way to not having to execute the source code every time you wanna convert a song, so I was thinking why it just worked from the source code, and I did a bit of researching, and it seems like when you just execute your program from the source code, its slightly different from when you execute the same program but with a executable, so I wanted to know what the differences were between executing from source code and executing from the executable, and I remembered that the program is built with a tool called `pyinstaller` , so I went to their page to read the docs, and it seems like there are several ways to build the program, the default way is just run `pyinstaller main.py` but I think you have to add a windowed flag, so that way it runs properly I think, but I'm not sure honestly, so it would be something like `pyinstaller -w main.py`, and if you only run that, it would be built by default, and so, when it's built, it will look somthing like this: 
![Screenshot_160](https://user-images.githubusercontent.com/58754180/135797006-dae631e8-dcd7-4a2f-a7d0-7098b4a87355.png)
but in the release you saw that is only an executable and the `config.json` file, that's because he added an extra flag, so he ran `pyinstaller -F -w main.py` and what the `-F` flag does, is that instead of creating all those bunch of files, it only creates one file which is the exe file, and he just added the config.json file, and for some reason, with the `-F` flag doesn't work, so, I tried to build but without it, and it works, so maybe has something to do with how pyinstaller works?

**Comment by Corne2Plum3** [comment]
I followed the suggestion of @Marcos875433 which consists to remove `-F`  when exporting the program. It works fine but there is more file in the exported version and ffmpeg windows are visible, but this is not a problem. I prefer having a program with these negligible issues which works instead of a program that doesn't work at all...

I released the **1.1.1** version with these fixes.

Thanks guys ^^
